lib/std/meta.zig

```diff
@@ -556,6 +556,32 @@ test "std.meta.fieldInfo" {
     try testing.expect(comptime uf.type == u8);
 }
 
+pub fn FieldType(comptime T: type, comptime field: FieldEnum(T)) type {
+    if (@typeInfo(T) != .Struct and @typeInfo(T) != .Union) {
+        @compileError("Expected struct or union, found '" ++ @typeName(T) ++ "'");
+    }
+
```

Comment:
this is superfluous as `fieldInfo` already does the check

Reply:
No, `fieldInfo` also allows error sets and enums.